test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -1339,6 +1342,242 @@ defmodule Trento.SapSystemTest do
   end
 
   describe "deregistration" do
+    test "should deregister a single DB instance if no SR enabled" do
+      sap_system_id = UUID.uuid4()
+      deregistered_at = DateTime.utc_now()
+
+      instances =
+        [
+          %DatabaseInstanceRegistered{
+            instance_number: instance_number_1,
+            host_id: host_id
+          },
+          %DatabaseInstanceRegistered{
+            instance_number: instance_number_2
+          }
+        ] =
+        build_list(
+          2,
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          system_replication: nil
+        )
+
+      assert_events_and_state(
+        [
+          build(
```

Comment:
Is this the correct order of things? Or do we have `instance`, `database`, `instance` order?
Just to be more precise

Reply:
I did notice this and you are totally right. Putting it in the right order requires some changes to the factory (or at least to how we use it). Will give it another try.

Reply:
Actually @arbulu89, looking at database registration tests, we first emit DatabaseRegistered and then we emit the instances registered events. The flow chart confirms this.